lib/discourse_code_review/state/github_repo_categories.rb

```diff
@@ -36,7 +37,8 @@ def ensure_category(repo_name:, repo_id: nil)
           end
         end
 
-        if !category
+        # create new category only when repo_id is present
+        if category.blank? && repo_id.present?
```

Comment:
It might be better to merge this whole `if` as an `else` to an `if` block on line 33. You won't need to check `repo_id.present?` again, and this whole part is now executed only when the category wasn't found (on line 25) anyway so it makes sense for this code to live there 🙂

Reply:
Nice tip! Made it so.